### ISSUE TITLE
Perch2 has invalid torch device "gpu" as default and invalid default targets for the forward call

### ISSUE THREAD

**Issue by benmunkres** [kitzeslab/bioacoustics-model-zoo#50]
Setting up a fresh environment to run Perch2 inference I ran into a couple bugs - 

1. Instantiating Perch2 trying to use a gpu (eg `Perch2(device='gpu')`) throws a pytorch error since `torch.device('gpu')` was deprecated (I believe) - this can be fixed by changing 'gpu' to cuda and modifying the if statement selecting the Perch2 version (eg if `device == 'cuda'` instead of `device == 'gpu'`)

1.  The default targets for the forward method of Perch2 are `targets=("logits", "embeddings", "spatial_embeddings", "spectrograms")` , but the tensorflow model outputs a dict with keys as the singular version of each of these, eg targets should be `targets=("logit", "embedding", "spatial_embedding", "spectrogram")`
